zingo-sync/src/sync.rs

```diff
@@ -127,14 +176,84 @@ fn prepare_next_scan_range(sync_state: &SyncState) -> Option<ScanRange> {
     if let Some((lower_range, higher_range)) = selected_scan_range
         .split_at(selected_scan_range.block_range().start + BlockHeight::from_u32(BATCH_SIZE))
     {
-        scan_ranges.splice(index..=index, vec![lower_range.clone(), higher_range]);
-
-        // TODO: set selected scan range to `ignored` so the same range is not selected twice when multiple tasks call this fn
+        let lower_range_ignored =
+            ScanRange::from_parts(lower_range.block_range().clone(), ScanPriority::Ignored);
+        scan_ranges.splice(index..=index, vec![lower_range_ignored, higher_range]);
 
         Some(lower_range)
     } else {
-        // TODO: set selected scan range to `ignored` so the same range is not selected twice when multiple tasks call this fn
+        let selected_scan_range = selected_scan_range.clone();
+        let selected_range_ignored = ScanRange::from_parts(
+            selected_scan_range.block_range().clone(),
+            ScanPriority::Ignored,
+        );
+        scan_ranges.splice(index..=index, vec![selected_range_ignored]);
 
         Some(selected_scan_range.clone())
     }
 }
+
+fn mark_scanned(scan_range: ScanRange, sync_state: &mut SyncState) -> Result<(), ()> {
+    let scan_ranges = sync_state.scan_ranges_mut();
+
+    if let Some((index, range)) = scan_ranges
```

Comment:
there has gotta be a better way to do this! eventually

not just because of mark. but because == comparison on the range seems weird, and because rebuilding it instead of changing its priority seems weird, and because panicking is not great, and because what if the ranges overlappp?

Reply:
i am open to suggestions about the comparison but i dont see an issue, it is essentially just two numbers that we must make sure are identical.

i agree rebuilding to change priority could be improved, again we will have to open a PR into LRZ or copy it into our lib.

in terms of overlapping. only one scan range is ever created, from birthday to chain height. then this scan range is only split so no overlapping or gaps are possible, as long as we unit test the split and splicing functions properly. the only exception is when ranges are rebuilt to change its priority but the ranges are copied so we can assume this will never be bugged and should also be tested eventually